taskchampion/xtask/src/main.rs

```diff
@@ -3,17 +3,42 @@
 //! At the moment it is very simple, but if this grows more subcommands then
 //! it will be sensible to use `clap` or another similar library.
 
+//! The const MSRV_FILE_PATHS_REGEX in /xtask/main.rs is an array of tuples
+//!  of the form (PATH, REGEX) where PATH and REGEX are literals where
+//!  PATH is a file that updates the Minimum Supported Rust Version
+//!  and REGEX is the pattern to find the appropriate line in the file
+
+use regex::Regex;
 use std::env;
 use std::fs::File;
-use std::io::Write;
-use std::path::PathBuf;
+use std::io::{BufRead, BufReader};
+use std::io::{Seek, Write};
+use std::path::{Path, PathBuf};
+
+// Increment length of array when adding tuples of (PATH, REGEX).
+const MSRV_PATH_REGEX: [(&str, &str); 2] = [
+    (
+        "../../.github/workflows/rust-tests.yml",
+        r#"- "[0-9]+("|\.|[0-9])+"#,
+    ),
+    (
+        "../../.github/workflows/checks.yml",
+        r#"toolchain: "[0-9]+("|\.|[0-9])+"#,
+    ),
+];
 
 pub fn main() -> anyhow::Result<()> {
-    let arg = env::args().nth(1);
-    match arg.as_deref() {
-        Some("codegen") => codegen(),
-        Some(arg) => anyhow::bail!("unknown xtask {}", arg),
-        _ => anyhow::bail!("unknown xtask"),
+    let arguments: Vec<String> = env::args().collect();
+    if arguments.len() < 2 {
+        anyhow::bail!("xtask: Valid arguments are: `codegen`, `msrv <version x.y>`");
+    }
+
+    println!("{:#?}", env::current_dir());
```

Comment:
I ran this script in the root of the repository, and as you suggested the paths didn't work:
```
xtask: Warning: xtask: Path "../../.github/workflows/rust-tests.yml" not found. Skipping.
xtask: Warning: xtask: Path "../../.github/workflows/checks.yml" not found. Skipping.
```
However, I updated this line to print all env vars, and saw
```
"CARGO_MANIFEST_DIR", "/home/dustin/p/taskwarrior/taskchampion/xtask"
```
which is the directory your paths are relative to. Perhaps you can use that environment variable to make this work regardless of directory?

Reply:
Oh man! This was a big "DERP!" on my part. Not only did you mention environment variables before, but the `codegen()` function literally uses the code already. /facepalm Sorry you had to spell it out for me.